x-pack/plugins/observability/public/pages/slo_edit/components/common/query_builder.tsx

```diff
@@ -74,6 +79,7 @@ export function QueryBuilder({
             disableLanguageSwitcher
             indexPatterns={dataView ? [dataView] : []}
             isDisabled={!indexPatternString}
+            isInvalid={fieldState.invalid}
```

Comment:
You also need to set the `isInvalid` field in the EuiFormRow above. So you need to use the `useFormContext()` and the `getFieldState` to access the invalid state, e.g.  `isInvalid={getFieldState(name).invalid}`